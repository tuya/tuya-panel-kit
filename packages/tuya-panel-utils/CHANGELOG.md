# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.1.0](https://github.com/tuya/tuya-panel-kit/compare/tuya-panel-utils@0.0.2...tuya-panel-utils@0.1.0) (2021-08-11)


### Features

* **Utils:** support Horizontal Screen ([#112](https://github.com/tuya/tuya-panel-kit/issues/112)) ([333292b](https://github.com/tuya/tuya-panel-kit/commit/333292be723e91136857979018d4795dc3d16ec1))





## [0.0.2](https://github.com/tuya/tuya-panel-kit/compare/tuya-panel-utils@0.0.1...tuya-panel-utils@0.0.2) (2021-07-29)

**Note:** Version bump only for package tuya-panel-utils





## 0.0.1 (2021-07-20)

**Note:** Version bump only for package tuya-panel-utils
