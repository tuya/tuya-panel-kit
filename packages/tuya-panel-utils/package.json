{
  "name": "tuya-panel-utils",
  "version": "0.2.3-rn72.0",
  "description": "Tuya tool collection kit",
  "author": "ZengBeauty<2497752603@qq.com>",
  "homepage": "https://github.com/tuya/tuya-panel-kit#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tuya/tuya-panel-kit.git"
  },
  "scripts": {
    "prebuild": "rimraf lib/*",
    "build": "tsc -d",
    "dev": "tsc -w",
    "test": "NODE_ENV=testing TZ=UTC jest -u --silent --forceExit",
    "test:watch": "jest --watch",
    "prepublishOnly": "yarn build"
  },
  "bugs": {
    "url": "https://github.com/tuya/tuya-panel-kit/issues"
  },
  "peerDependencies": {
    "react": "^16.8.3",
    "react-native": "^0.59.10"
  },
  "gitHead": "010cba9f5a3136808203e5a342bcd164601063e7"
}
