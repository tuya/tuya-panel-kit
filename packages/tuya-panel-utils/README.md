# `tuya-panel-utils`

Tuya tool collection kit

## Usage

```jsx
import { Utils } from 'tuya-panel-utils';
```
