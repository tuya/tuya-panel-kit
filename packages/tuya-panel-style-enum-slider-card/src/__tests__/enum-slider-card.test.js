/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Utils } from 'tuya-panel-utils';
import { AcrylicEnumSliderCard } from '../index'
import { mount } from 'enzyme';

const { ThemeProvider } = Utils.ThemeUtils;

const data = [
  {
    label: '关闭',
    key: '0',
  },
  {
    label: '一档',
    key: '1',
  },
  {
    label: '二档',
    key: '2',
  },
  {
    label: '三档',
    key: '3',
  },
];

describe('EnumSliderCard', () => {
  it('Acrylic', () => {
    const wrap = renderer.create(
      <ThemeProvider>
        <AcrylicEnumSliderCard
          data={data}
          title="AcrylicEnumSliderCard"
          bottomPromptTexts={['Off', 'Max']}
        />
      </ThemeProvider>
    ).toJSON();
    expect(wrap).toMatchSnapshot();
  });

  it('activeKey', () => {
    let activeKey = '1';
    const handSlidingComplete = (key, index) => activeKey = key;
    const wrap = mount(
      <ThemeProvider>
        <AcrylicEnumSliderCard
          data={data}
          title="AcrylicEnumSliderCard"
          bottomPromptTexts={['Off', 'Max']}
          handSlidingComplete={handSlidingComplete}
        />
      </ThemeProvider>
    );
    const sliderInstance = wrap.find('Slider');  
    sliderInstance.props().onSlidingComplete(3);
    wrap.update();
    expect(activeKey).toBe('3');
    expect(wrap.find('Slider').props().value).toBe(3);
  });
});