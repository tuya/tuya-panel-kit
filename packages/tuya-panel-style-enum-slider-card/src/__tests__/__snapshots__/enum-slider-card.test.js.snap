// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnumSliderCard Acrylic 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 32,
        "paddingBottom": 42,
        "paddingLeft": 40,
        "paddingRight": 40,
        "paddingTop": 42,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 48,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={32}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 32,
                },
                Array [
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontSize": 32,
                    "fontWeight": undefined,
                    "marginLeft": 0,
                  },
                  undefined,
                ],
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        AcrylicEnumSliderCard
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={32}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 32,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.5)",
                  "fontSize": 32,
                  "fontWeight": undefined,
                },
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        关闭
      </Text>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Slider"
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
            },
            Object {
              "flexDirection": "column",
              "height": 40,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.05)",
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 2,
                "height": 4,
              },
              Object {
                "borderRadius": 12,
                "height": 12,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 0,
                "height": 0,
                "width": 0,
              },
              Object {
                "backgroundColor": "#FFF",
                "justifyContent": "center",
                "position": "absolute",
              },
              Object {
                "left": 0,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.7)",
                "borderRadius": 8,
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 0,
                "height": 0,
                "width": 0,
              },
              Object {
                "backgroundColor": "#FFF",
                "justifyContent": "center",
                "position": "absolute",
              },
              Object {
                "left": 0,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.7)",
                "borderRadius": 8,
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 0,
                "height": 0,
                "width": 0,
              },
              Object {
                "backgroundColor": "#FFF",
                "justifyContent": "center",
                "position": "absolute",
              },
              Object {
                "left": 0,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.7)",
                "borderRadius": 8,
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 0,
                "height": 0,
                "width": 0,
              },
              Object {
                "backgroundColor": "#FFF",
                "justifyContent": "center",
                "position": "absolute",
              },
              Object {
                "left": 0,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.7)",
                "borderRadius": 8,
                "height": 8,
                "width": 8,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FE7862",
              "borderRadius": 12,
              "height": 12,
              "justifyContent": "center",
              "opacity": 0,
              "overflow": "hidden",
              "position": "absolute",
            }
          }
        />
        <View
          onLayout={[Function]}
          renderToHardwareTextureAndroid={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 48,
              "elevation": 2,
              "height": 48,
              "opacity": 0,
              "position": "absolute",
              "shadowColor": "rgba(0, 0, 0, 0.15)",
              "shadowOffset": Object {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 8,
              "width": 48,
            }
          }
        />
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 9,
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={24}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.45)",
                  "fontSize": 24,
                  "fontWeight": "normal",
                },
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        Off
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={24}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#333",
                  "fontSize": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.45)",
                  "fontSize": 24,
                  "fontWeight": "normal",
                },
              ],
            ],
          ]
        }
        type={null}
        weight={null}
      >
        Max
      </Text>
    </View>
  </View>
</View>
`;
